### ISSUE TITLE
Latest Update 

### ISSUE THREAD

**Issue by clburlison** [brandonusher/Apple-ID-AppleScript#54]
Hey everyone,

I was able to get this running on Friday (Sept 9) and Monday (Sept 12) however it seems like today Apple might have changed something on their end? I'm not sure why this is happening but the script now stops at the agree to terms and condition window. 

The AppleScript error message is:

    System Events got an error: Can’t get UI element 
    "Terms and Conditions and Apple Privacy Policy" of UI 
    element 1 of scroll area 1 of group 1 of group 1 of 
    splitter group 1 of window "iTunes" of application 
    process "iTunes".

Anyone have any ideas?  

iTunes: 12.4.3.1
OS: 10.11.6

Thanks,

**Comment by BillCampbell** [comment]
@clburlison 
Since iOS 10 released on Tuesday, Sept 13 (in the US at least), I'm guessing something changed with the terms windows and broke the script.  While I've found this script useful, this is why it is no longer maintained.  I have no reason to use it until at least next summer (and maybe not again ever), and I don't really have time to look at it.  However, tools I found useful over the course of a weekend and a week or so to resurrect the script this past summer were:

1. AppleScript Tutorial at Lynda.com (since I'd never used AppleScript before)
1. [AppleScript Language Guide from Apple](https://developer.apple.com/library/content/documentation/AppleScript/Conceptual/AppleScriptLangGuide/introduction/ASLR_intro.html)

1. [Old UI Script article from MacTech](http://www.mactech.com/articles/mactech/Vol.21/21.06/UserInterfaceScripting/index.html) (Again, this was new to me.)

1. Trial version of [UI Browser](http://pfiddlesoft.com/uibrowser/) by PFiddlesoft - To figure out the names of the UI elements.

1. Trial version of Script Debugger 6 by [Late Night Software](http://latenightsw.com/) - useful IDE for AppleScript

My trial versions of the tools above, which I found pretty essential, have expired so even if I had the time to jump back into this at the moment, which I don't, I don't have the tools, and I don't do this enough to license the software. (This was the first time I used AppleScript in a complex way and I haven't since.)

If you decide to try to fix the script, good luck and please merge the code back into this project!

Best regards,
Bill

**Comment by rmiguens** [comment]
Hi Bill,

	I agree - those who needed this should be moving towards Apple School Manger and such. Thank you for your contributions as this has helped us out once or twice!
Thanks,

Roy Miguens 
Director of IT
Calgary Arts Academy
The Art of Learning
(403) 229-3010 ext 2228

Calgary Arts Academy is an innovative charter school that engages children through Arts Immersion to become confident learners and spirited citizens.

Confidentiality Notice – The information in this email, including any attached documents is confidential and is intended solely for the named recipient(s). Any distribution, use or copying of this email and the information it contains by other than an intended recipient is unauthorized. If you received this email in error, please advise the sender immediately.

**Comment by clburlison** [comment]
@BillCampbell Thanks for the resources. If I'm able to find time and get this working I'll definitely create a PR.

**Comment by seesolve** [comment]
Highly recommend option 4. PFiddlesoft's UI Browser can easily help you identify the correct reference to the UI element.

**Comment by jploughe** [comment]
The only problem with the accounts created in ASM is that they can only browse the store, they cannot get apps.  If you want the ID's to still be able to get free apps, you have to create the ID by hand.

**Comment by BillCampbell** [comment]
@jploughe 
The idea behind using Managed Apple Ids created in ASM is that you would distribute apps using an MDM or maybe Apple Configurator.  (I use JAMF Casper so I don't know much about Configurator.)  With Casper I can "purchase" app licenses (even for free app) via the Apple Volume purchase store and distribute those apps via my own Self Service portal. Therefore, the App Store and making purchases is not required.  

That said, another problem with Managed Apple IDs, which is one reason I didn't use them this past summer, is that Find My iPad is not supported.  While the JAMF MDM does allow you to put an iPad in lost mode and will give you a location using iPad GPS data, that feature in the MDM does not allow you to activate an audible ping on the device. Our IT staff and students currently use that feature a lot so we avoided Managed Apple IDs this past summer, which is why I hacked at this script to make it work.  My hope is that before next summer Apple or JAMF will deal with that somehow making Managed Apple IDs a better option and this script no longer necessary for my deployment.

Another possible problem is that Managed Apple IDs currently (maybe this will change) only provide the 5 GB of free iCloud storage, which we use for backups.  There is no way to purchase more storage for a Managed Apple ID at the time I'm posting this as far as I can tell.

**Comment by jploughe** [comment]
@BillCampbell 
I agree that in an ideal world, they should only be getting apps via our MDM.  However,  those with higher pay grades than me say the users should be able to get free apps of their choosing when they want them. 
I do hope someone can fix the script to get past the Agree to Terms page.  I don't know enough about apple scripting yet to understand its conventions.
